jasmine-standalone-3.5.0/src/tenPinApp.js

```diff
@@ -0,0 +1,43 @@
+function Bowling () {
+this.player_score = 0;
+this.strike = false;
+this.spare = false;
+this.frame = 0;
+this.bonus = false;
+}
+
+Bowling.prototype.calcPoints = function (r1,r2){
+
+    if (r1 == 10){
+        this.strike = true;
+        this.player_score +=10;   
+    }
+     else if (this.strike == true){
+        this.player_score += (r1 + r2) * 2;
+        this.strike = false;
+    }
+    else if (this.spare == true){
+        this.spare = false;
+        this.player_score += (r1*2)+r2
+    }
```

Comment:
this method is a little bit long. You have the conditionals and then doing 2 extra things within each conditional. Does this follow SRP?

Reply:
Just because it's long doesn't mean it doesn't follow SRP. SRP means doing one thing, but it can be a complex thing, as long as it is serving one purpose (here adding up the score).